src/ica_api.c

```diff
@@ -1367,8 +1367,12 @@ int ica_ec_key_init(const unsigned char *X, const unsigned char *Y,
 #ifdef ICA_FIPS
 	if (fips >> 1)
 		return EACCES;
-	if (!curve_supported_via_openssl(key->nid) || !curve_supported_via_cpacf(key->nid))
-		return EPERM;
+	if (fips & ICA_FIPS_MODE) {
```

Comment:
For all these ICA_FIPS_MODE tests (here and in the following commits); Why not combining them into one if? AFAIK there is only one single path.

Reply:
Yes, the two checks curve_supported_via_openssl() and curve_supported_via_cpacf() are used at several places together. Will check to consolidate this after this pr.